pkg/handlers/v1/ec2.go

```diff
@@ -114,24 +196,39 @@ func (t ec2Transformer) Update(event awsConfigEvent) (Output, error) {
 		changes.PrivateIPAddresses = append(changes.PrivateIPAddresses, private...)
 		changes.PublicIPAddresses = append(changes.PublicIPAddresses, public...)
 		changes.Hostnames = append(changes.Hostnames, dns...)
+		// collect added IPs & hostnames and their associated attachTimes, for later EC2 event splitting
+		if diff.ChangeType != delete {
+			for _, privateIP := range private {
+				addPrivateIPMap[privateIP] = ni.Attachment.AttachTime
+			}
+			for _, publicIP := range public {
+				addPublicIPMap[publicIP] = ni.Attachment.AttachTime
+			}
+			for _, hostname := range dns {
+				addHostnameMap[hostname] = ni.Attachment.AttachTime
+			}
+		}
 	}
 
 	// We need to compute the symmetric difference of the added changes and the removed changes
 	// i.e. remove entries that show up as both added and removed
 	symmetricDifference(&addedChange, &deletedChange)
 	if len(addedChange.PrivateIPAddresses) > 0 || len(addedChange.PublicIPAddresses) > 0 || len(addedChange.Hostnames) > 0 {
-		output.Changes = append(output.Changes, addedChange)
+		addedOutputs := separateAddedChanges(baseOutput, addedChange, addPrivateIPMap, addPublicIPMap, addHostnameMap)
+		outputs = append(outputs, addedOutputs...)
 	}
 	if len(deletedChange.PrivateIPAddresses) > 0 || len(deletedChange.PublicIPAddresses) > 0 || len(deletedChange.Hostnames) > 0 {
-		output.Changes = append(output.Changes, deletedChange)
+		deletedOutput := baseOutput
+		deletedOutput.Changes = append(deletedOutput.Changes, deletedChange)
+		outputs = append(outputs, deletedOutput)
 	}
-	return output, nil
+	return outputs, nil
 }
 
-func (t ec2Transformer) Delete(event awsConfigEvent) (Output, error) {
+func (t ec2Transformer) Delete(event awsConfigEvent) ([]Output, error) {
```

Comment:
Note for reviewers: We are still using configurationItemCaptureTime for the deletion events because we don't currently have any other options. That's why this method isn't changing much.